File: extensions/positron-python/uitests/uitests/steps/status_bar.py
# Copyright (c) Microsoft Corporation. All rights reserved.
# Licensed under the MIT License.


import behave
from selenium.common.exceptions import StaleElementReferenceException

import uitests.vscode.status_bar
import uitests.tools


@behave.then(
    'the python interpreter displayed in the the status bar contains the value "{name}" in the tooltip'
)
@uitests.tools.retry((AssertionError, StaleElementReferenceException))
def then_selected_interpreter_has_tooltip(context, name):
    element = uitests.vscode.status_bar.wait_for_python_statusbar(context)
    title = element.get_attribute("title")
    assert name in title, f"{name} in {title}"


@behave.then(
    'the python interpreter displayed in the the status bar contains the value "{name}" in the display name'
)
@uitests.tools.retry((AssertionError, StaleElementReferenceException))
def then_selected_interpreter_has_text(context, name):
    element = uitests.vscode.status_bar.wait_for_python_statusbar(context)
    assert name in element.text, f"{name} not in {element.text}"


@behave.then(
    'the python interpreter displayed in the the status bar does not contain the value "{name}" in the display name'
)
@uitests.tools.retry((AssertionError, StaleElementReferenceException))
def then_selected_interpreter_does_not_have_text(context, name):
    element = uitests.vscode.status_bar.wait_for_python_statusbar(context)
    assert name not in element.text, f"{name} in {element.text}"

File: extensions/positron-python/uitests/src/steps/statusbar.ts
// Copyright (c) Microsoft Corporation. All rights reserved.
// Licensed under the MIT License.

'use strict';

// tslint:disable: no-invalid-this

import { expect } from 'chai';
import { Given, Then, When } from 'cucumber';
import { CucumberRetryMax5Seconds } from '../constants';
import { noop } from '../helpers';
import '../helpers/extensions';

Given('the python status bar item is hidden', async function() {
    await this.app.statusbar.hidePythonStatusBarItem().catch(noop);
});
When('I hide the python status bar item', async function() {
    await this.app.statusbar.hidePythonStatusBarItem().catch(noop);
});
Then('hide the python status bar item', async function() {
    await this.app.statusbar.hidePythonStatusBarItem().catch(noop);
});

// Add a delay, as this can take around 1s (from the time something was selected).
Then('the python the status bar contains the text {string}', CucumberRetryMax5Seconds, async function(text: string) {
    const statubarText = await this.app.statusbar.getPythonStatusBarText();
    expect(statubarText).contains(text);
});

// Add a delay, as this can take around 1s (from the time something was selected).
Then('the python the status bar does not contain the text {string}', CucumberRetryMax5Seconds, async function(text: string) {
    const statubarText = await this.app.statusbar.getPythonStatusBarText();
    expect(statubarText).not.contains(text);
});

Then('a status bar item containing the text {string} is displayed', async function(text: string) {
    await this.app.statusbar.waitUntilStatusBarItemWithText(text);
});

// Then('the python the status bar is not visible', async () => {
//     await context.app.workbench.statusbar.pythonStatusBarElementIsNotVisible();
// });
